internal/pkg/cli/pipeline_delete.go

```diff
@@ -63,42 +72,50 @@ func newDeletePipelineOpts(vars deletePipelineVars) (*deletePipelineOpts, error)
 	if err != nil {
 		return nil, fmt.Errorf("default session: %w", err)
 	}
+	ssmStore := config.NewSSMStore(identity.New(defaultSess), ssm.New(defaultSess), aws.StringValue(defaultSess.Config.Region))
+	prompter := prompt.New()
 
 	opts := &deletePipelineOpts{
 		deletePipelineVars: vars,
 		prog:               termprogress.NewSpinner(log.DiagnosticWriter),
-		prompt:             prompt.New(),
+		prompt:             prompter,
+		sel:                selector.NewConfigSelect(prompter, ssmStore),
 		secretsmanager:     secretsmanager.New(defaultSess),
 		pipelineDeployer:   cloudformation.New(defaultSess),
 		ws:                 ws,
+		store:              ssmStore,
 	}
 
 	return opts, nil
 }
 
 // Validate returns an error if the flag values passed by the user are invalid.
 func (o *deletePipelineOpts) Validate() error {
-	if o.appName == "" {
-		return errNoAppInWorkspace
-	}
+	return nil
+}
 
+// Ask prompts for and validates required fields.
+func (o *deletePipelineOpts) Ask() error {
+	if o.appName != "" {
+		if _, err := o.store.GetApplication(o.appName); err != nil {
+			return err
+		}
+	} else {
+		if err := o.askAppName(); err != nil {
+			return err
+		}
+	}
 	if o.name != "" {
 		if _, err := o.codepipeline.GetPipeline(o.name); err != nil {
 			return err
 		}
 	}
```

Comment:
If name is not empty should we ask users to select from a list? or that's the next step after this PR

Reply:
Yup, that's the next step. 👍🏼